### ISSUE TITLE
Issues with yt_new / v0.18.5

### ISSUE THREAD

**Issue by B0pol** [TeamNewPipe/NewPipeExtractor#263]
Hi,
I've updated NewPipe to v0.18.5 and noticed issues:

~~Description:~~
+ ~~timestamps are broken (you can see with [this video](https://www.youtube.com/watch?v=-FHpYUuYrcQ)). Note: Regression, it just fixed in 0.18.4, only for HTML formatting.
Maybe you should use the old form, which was ``<a href="https://www.youtube.com/watch?v=-FHpYUuYrcQ&amp;t=1235">20:35</a>``, and maybe you want to see https://github.com/TeamNewPipe/NewPipeExtractor/pull/188 which fixed timestamps in NewPipeExtractor long ago.~~
+ ~~(Some?) channel links are broken (you can see with [this video](https://www.youtube.com/watch?v=srnxL6KX4n8)). Maybe it's not for all channels, I haven't figured out. Anyway on Youtube website, it's a hyperlink ``@CHANNEL``.~~
+ ~~(Some?) video links are broken (you can see with [this video](https://www.youtube.com/watch?v=hrLrEfP2Wjo), CTRL+F "fond", under "musique"). Or also [here](https://www.youtube.com/watch?v=lmdjyU1YVLw), CTRL+F for "Episode précedent".~~

Fixed in #262 

~~Trending:
Only the first three videos are shown. Already mentionned in https://github.com/TeamNewPipe/NewPipe/issues/3145.~~
Fixed in #262 

~~Titles:
On videoDetailDescription (then YoutubeStreamExtractor), (some?) channels aren't supported. In search it work though. Cf [this video](https://www.youtube.com/watch?v=srnxL6KX4n8). Related to the issue with channels in description above.~~
Fixed in #262 

~~Profile pictures: see below or https://github.com/TeamNewPipe/NewPipe/issues/3144~~
Fixed in #262 

~~It shows "subscribers count unavailable" instead of "no subscribers" for channels with 0 subs.~~
Fixed in #262 

~~Newlines are given instead of hashtags at the end of a video. You can reproduce with this video:
https://www.youtube.com/watch?v=srnxL6KX4n8~~

Remaining issues:
…

I will edit if I found more issues.

**Comment by PeterHindes** [comment]
[This bug](https://github.com/TeamNewPipe/NewPipe/pull/2724) has also been reintroduced in the latest update.

**Comment by B0pol** [comment]
Wrong profile pictures sometimes. You car try with UK trendings:
<img src="https://i.ibb.co/jL8dgk7/Screenshot-20200226-204358-New-Pipe.png" width="200"> <img src="https://i.ibb.co/jzPF2FN/Screenshot-20200226-204401-New-Pipe.png" width="200">
<img src="https://i.ibb.co/ySKB16N/Screenshot-20200226-204409-New-Pipe.png" width="200"> <img src="https://i.ibb.co/4Pg0cDp/Screenshot-20200226-204412-New-Pipe.png" width="200">
I first thought is was because trending page is broken but it also happens to some of my subscriptions.

**Comment by PeterHindes** [comment]
opened issue #264

**Comment by wb9688** [comment]
@B0pol: Already reported at TeamNewPipe/NewPipe#3144.

**Comment by B0pol** [comment]
+ randomly hit channel page from people in comments, and I got this error.
It seems to happen on all channels without videos.

## Exception
* __User Action:__ requested channel
* __Request:__ https://www.youtube.com/channel/UCCeHFuWgQb547eDKBwVm7hg
* __Content Language:__ FR
* __Package:__ org.schabi.newpipe.debug
* __Version:__ 0.18.5
* __Service:__ YouTube
* __Operating System:__ Linux Android 8.1.0 - 27


<details><summary><b>Crash log</b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'com.grack.nanojson.JsonArray com.grack.nanojson.JsonObject.getArray(java.lang.String)' on a null object reference
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeChannelExtractor.getInitialPage(YoutubeChannelExtractor.java:200)
	at org.schabi.newpipe.extractor.utils.ExtractorHelper.getItemsPageOrLogError(ExtractorHelper.java:19)
	at org.schabi.newpipe.extractor.channel.ChannelInfo.getInfo(ChannelInfo.java:82)
	at org.schabi.newpipe.extractor.channel.ChannelInfo.getInfo(ChannelInfo.java:47)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getChannelInfo$4(ExtractorHelper.java:123)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$u5W7VszTe8AoEexIsFM9huQfbkM.call(Unknown Source:4)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.Flowable.subscribe(Flowable.java:14479)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)

```
</p></details>
<hr>

+ channel description are cut in search page

(I using debug build based on newpipe dev and your latest commit in #262)

**Comment by wb9688** [comment]
@B0pol: I could confirm that exception. It doesn't happen in v0.18.5 though. I'll probably fix it tomorrow, just like some of the other reported bugs.

**Comment by wb9688** [comment]
@B0pol: I just fixed a bunch of stuff.

**Comment by B0pol** [comment]
It says "subscribers count unavailable" when people have 0 sub. With 0.18.4, it was showing "no subscribers"
You can reproduce with [this channed](https://youtube.com/channel/UC2TkMR76WPwwBzUMaVc99xw)
I reproduced with both latest lebug amd official 0.18.5

**Comment by wb9688** [comment]
@B0pol: I noticed that as well, but forgot about it. I'll fix that in ±5 hours.

**Comment by artsemionov** [comment]
Don't know if reported, but search isn't working when using in https://github.com/yausername/NewPipeREST
I get org.schabi.newpipe.extractor.exceptions.ParsingException: Could not get ytInitialData

**Comment by wb9688** [comment]
@B0pol: What do you mean with missing playlist descriptions?

**Comment by B0pol** [comment]
I don't know. Was this once in NewPipe? I remember but I'm not sure now. I'm talking about online playlist, they have a description, and I thought in NewPipe they where shown only on two lines, below title, with ellipsize … if it's more than 2 lines, but I am not sure now.
For example, the description for [this playlist](https://www.youtube.com/playlist?list=PLD7SPvDoEddYd3OLS9T2ZT4eGc1iTwYG4) is 
```
UK Top 40 This Week 2020 - Charts 2020 Playlist - Top 10 Songs This Week 2020
=================================================================
Update every week
===============================================================
Good music calms my soul, it takes me to another place, away from my problems
===============================================================

This are the official UK top 100 singles charts for this week of january 2019, february 2019, march 2019, april 2019, may 2019, june 2019, july 2019, august 2019, september 2019, october 2019, november 2019, december 2019, january 2020
 
We will continue to update this playlist also for this month uk top 40 singles charts this week of 2019 and other months constantly.
```

**Comment by artsemionov** [comment]
In 0.18.5 you can't get next page for search in projects like NewPipeRest. As I understand now it depends on the first search, if you call getPage separately you get NPE because 'initialData' and 'doc' are null.

**Comment by wb9688** [comment]
@B0pol: NewPipeExtractor doesn't have any methods for extracting the description of a playlist. Create a new issue for that and I'll add it once #262 is finished and merged.

**Comment by wb9688** [comment]
@B0pol: YouTube doesn't distinguish between '0 subscribers' and 'subscriber count unavailable' anymore.

@TobiGr: Do you think we should say '0 subscribers' or 'subscriber count unavailable' in case there's no subscriber count?

**Comment by TobiGr** [comment]
I prefer unavailable, because we do not have information about it.

**Comment by wb9688** [comment]
@TobiGr: OK, then I'll keep it that way. @B0pol: In that case, that's not a bug.

**Comment by wb9688** [comment]
@TobiGr: OK, then I'll keep it that way. @B0pol: In that case, that's not a bug.

**Comment by wb9688** [comment]
@TobiGr: OK, then I'll keep it that way. @B0pol: In that case, that's not a bug.

**Comment by wb9688** [comment]
SrainUser on IRC pointed out that if there is a subscribe button, we could say "0 subscribers", and if there's no subscribe button, we could say "subscriber count unavailable".

**Comment by B0pol** [comment]
I found another « bug »
It shows newlines instead of hashtags at the end of descriptions
https://www.youtube.com/watch?v=srnxL6KX4n8

**Comment by wb9688** [comment]
@B0pol: Could you please test my latest commit? Your latest bug shouldn't happen there.

**Comment by B0pol** [comment]
yeah, it's fixed
